### PR TITLE
Detect integers 6644 v6

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6644 and all subtickets
https://redmine.openinfosecfoundation.org/issues/6645
https://redmine.openinfosecfoundation.org/issues/6646
https://redmine.openinfosecfoundation.org/issues/6647
https://redmine.openinfosecfoundation.org/issues/6648
https://redmine.openinfosecfoundation.org/issues/6628

Describe changes:
- detect/integers: support hexadecimal notation for parsing
- detect/integers: add mode for negated range
- detect/integers: rust derive for enumerations
- detect/integers: keywords now accept bitmasks
- doc: detect/integers

https://github.com/OISF/suricata/pull/10122 with review taken into account

### CODE REVIEW

**Review by inashivb — COMMENTED** [review]
This is excellent work.
A lot of Rust generics code was difficult to understand for me and I could have used comments in the code to understand more easily esp `derive_enum_string`.
I'm excited for the use of this in actual rules.

Q: While the unittests work well, do you think we should modify some existing s-v tests to have some examples there as well?

**Comment by jufajardini** [comment]
> This is excellent work. A lot of Rust generics code was difficult to understand for me and I could have used comments in the code to understand more easily esp `derive_enum_string`. I'm excited for the use of this in actual rules.
> 
> Q: While the unittests work well, do you think we should modify some existing s-v tests to have some examples there as well?

+1 for everything, but imagining that the SV tests examples could be a separate ticket? 🤔

**Comment by catenacyber** [comment]
SV tests is kind of a chicken and egg problem :
- I created these features for some keywords that are needed in other PRs like websockets
- These PRs have SV tests
- I separated the feature as it benefits other integer keywords... but there are no relevant SV tests for it

File: rust/src/detect/uint.rs [hunk 1]
```diff
@@ -153,12 +161,17 @@ fn detect_parse_uint_start_interval_inclusive<T: DetectIntType>(
     let (i, arg2) = verify(detect_parse_uint_value::<T>, |x| {
         *x > arg1 && *x < T::max_value()
     })(i)?;
+        let mode = if neg.is_some() {
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply rust/src/detect/uint.rs:218]: fixing, thanks

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/10151

**Pull request closed without merging**